gateway/ln-gateway/src/rpc/mod.rs

```diff
@@ -34,11 +35,20 @@ impl GatewayRpcSender {
     {
         let (sender, receiver) = oneshot::channel::<Result<R::Response>>();
         let msg = message.to_enum(sender);
-        self.sender
-            .send(msg)
+
+        if let Err(e) = self.sender.send(msg).await {
+            error!("Failed to send message over channel: {}", e);
+            return Err(e.into());
+        }
+
+        receiver
             .await
-            .expect("failed to send over channel");
-        Ok(receiver.await.expect("Failed to send over channel")?)
+            .unwrap_or_else(|_| {
+                Err(LnGatewayError::Other(anyhow!(
+                    "Failed to receive response over channel"
+                )))
+            })
+            .map_err(|e| e.into())
```

Comment:
This code doesn't feel super idiomatic but I can't find any better way of expressing what it does. Maybe @dpc as an idea? Not a blocker, would just be neat to have in my mental toolbox.

Reply:
Superficially `.map_err(Into::into)` might be slightly better.

But maybe:

```rust
Ok(receiver.await.unwrap_or_else(...)?)
```

instead?